### PR TITLE
Bump alpine

### PR DESCRIPTION
**Pull request by wzshiming**
The alpine:20231219 has a critical severity CVE, upgrade to the latest to fix it!
 
https://hub.docker.com/layers/library/alpine/20231219/images/sha256-60eda2a7bc29a54fe6beae0d72312ea995eb3b8387535e8dbf6767fd1b765d34?context=explore

/cc @ameukam
/cc @airbornepony
/cc @listx

### CODE REVIEW

File: config/prow/cluster/build/create-loop-devs_daemonset.yaml [hunk 1]
```diff
@@ -44,7 +44,7 @@ spec:
             done
             sleep 100000000
           done
-        image: alpine:20231219
+        image: alpine:latest
```
Comment (ameukam) [inline config/prow/cluster/build/create-loop-devs_daemonset.yaml:47]:
```suggestion
        image: alpine:20240606
```

File: config/prow/cluster/build/tune-sysctls_daemonset.yaml [hunk 2]
```diff
@@ -36,7 +36,7 @@ spec:
             sysctl -w fs.inotify.max_user_watches=524288
             sleep 10
           done
-        image: alpine:20231219
+        image: alpine:latest
```
Comment (ameukam) [inline config/prow/cluster/build/tune-sysctls_daemonset.yaml:39]:
```suggestion
        image: alpine:20240606
```

File: images/alpine/Dockerfile [hunk 3]
```diff
@@ -15,7 +15,7 @@
 # Update the date below (check https://hub.docker.com/_/alpine/tags) to get the
 # alpine edge base image, which has a faster release cadence than alpine:latest
 # (which tracks alpine stable), to pick up security patches more quickly.
-FROM alpine:20231219
+FROM alpine:latest
```
Comment (ameukam) [inline images/alpine/Dockerfile:18]:
```suggestion
FROM alpine:20240606
```

File: images/git/Dockerfile [hunk 4]
```diff
@@ -15,7 +15,7 @@
 # Update the date below (check https://hub.docker.com/_/alpine/tags) to get the
 # alpine edge base image, which has a faster release cadence than alpine:latest
 # (which tracks alpine stable), to pick up security patches more quickly.
-FROM alpine:20231219
+FROM alpine:latest
```
Comment (ameukam) [inline images/git/Dockerfile:18]:
```suggestion
FROM alpine:20240606
```

**Comment by wzshiming** [comment]
Done

**Comment by wzshiming** [comment]
[alpine:latest](https://hub.docker.com/layers/library/alpine/latest/images/sha256-eddacbc7e24bf8799a4ed3cdcfa50d4b88a323695ad80f317b6629883b2c2a78?context=explore)
[alpine:20240606](https://hub.docker.com/layers/library/alpine/20240606/images/sha256-5f3877672d1e408d63126ebcd62fa84183a627955762876ac64315722fb7fa87?context=explore)

It looks like the `latest` is newer than the `edge`, and the current `edge` still seems to have a CVE.

**Pull request merged**